src/hypothesis/strategies.py

```diff
@@ -927,11 +927,18 @@ def from_type(thing):
     """
     from hypothesis.searchstrategy import types
     if not isinstance(thing, type):
-        # Under Python 3.6, Unions are not instances of `type` - but we still
-        # want to resolve them!  This __origin__ check only passes if thing is
-        # a Union with parameters; if it doesn't we can't resolve it anyway.
         try:
+            # At runtime, `typing.NewType` returns the identity function rather
+            # than an actual type, but we can check that for a possible match
+            # and then read the magic attribute to unwrap it.
             import typing
+            if all([
```

Comment:
It doesn't matter overly, but why is this after the guard on being able to import typing?

Reply:
Because if we can't `import typing`, we don't need special handling for things in the `typing` module.  It's *mostly* redundant with the object `__module__` check I think, but I like it as a clear signal that we only try resolving NewType if typing is available.